sbi/inference/snpe/snpe_base.py

```diff
@@ -435,6 +420,26 @@ def _train(
         self._summary["epochs"].append(epoch)
         self._summary["best_validation_log_probs"].append(self._best_val_log_prob)
 
+    def _loss(
+        self, round_idx: int, theta_batch: Tensor, x_batch: Tensor, masks: Tensor
+    ) -> Tensor:
+        """Return the right calibrated loss, depending on the round."""
+
+        if round_idx == 0:
+            # Use plain posterior log prob for first round.
+            log_prob_batch = self._neural_posterior.neural_net.log_prob(
+                theta_batch, x_batch
+            )
+        else:
+            # Use proposal posterior log prob tailored to snpe version (B, C).
+            log_prob_batch = self._get_log_prob_proposal_posterior(
+                theta_batch, x_batch, masks
+            )
+
+        # Irrespective of round or snpe version, weight with calibration kernel.
+        # Return negative log prob to get the loss.
```

Comment:
I feel this should be in a 2nd line in the docstring - "The loss is the negative log prob" (though here David would argue: negative log-likelihood.)